### PR TITLE
Merge libraries in project

### PR DESCRIPTION
**Pull request by kokyhin**

### CODE REVIEW

File: src/ms3/ms3-to-oas/merge-library-to-ms3.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import * as MS3 from './../ms3-v1-api-interface';
+import { cloneDeep, forEach, map } from 'lodash';
+import { Library } from './../ms3-v1-library-interface';
+import { Trait } from './../ms3-v1-api-interface';
+
+class MergeLibraryToMs3 {
+  libraries: Library[];
+  constructor(private API: MS3.API) {}
+
+  checkDuplicates(targetArr: any, item: any, counter: number = 1): any {
```
Comment (nataliacr) [inline src/ms3/ms3-to-oas/merge-library-to-ms3.ts]: checkDuplicates(targetArr: any[], item: any, counter: number = 1): any[] {

File: src/ms3/ms3-to-oas/merge-library-to-ms3.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    const prop = item.name ? 'name' : 'title';
```
Comment (nataliacr) [inline src/ms3/ms3-to-oas/merge-library-to-ms3.ts]: rename prop to 'propName' or just 'name'

File: src/ms3/ms3-to-oas/merge-library-to-ms3.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    const traitNames = map(targetArr, (trait: any) => trait[prop]);
```
Comment (nataliacr) [inline src/ms3/ms3-to-oas/merge-library-to-ms3.ts]: why trait if this function is used for every api part? 
rename to smth appropriate

File: src/test/files/Project-with-library.ts [hunk 4]
```diff
@@ -0,0 +1,1322 @@
...
+  '_creator': '587620f13b4e9f19192177e4',
+  'lockedBy': 'alexei',
+  '_deleted': false,
+  'examples': [
+    {
+      'title': 'ProjectExample',
+      'description': '',
+      'content': '<name>My Name<\/name>',
+      'format': 'xml',
+      'annotations': [
```
Comment (nataliacr) [inline src/test/files/Project-with-library.ts:16]: you should test on already sanitized project, so remove any empty arrays, objects and strings

File: src/ms3/ms3-to-oas/merge-library-to-ms3.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      lib.resourcesTypes.forEach((resourceType) => this.checkDuplicates(project.resourcesTypes, resourceType));
+      lib.securitySchemes.forEach((secSchema) => this.checkDuplicates(project.securitySchemes, secSchema));
+      lib.annotationTypes.forEach((annotationType) => this.checkDuplicates(project.annotationTypes, annotationType));
+      lib.dataTypes.forEach((dataType) => this.checkDuplicates(project.dataTypes, dataType));
+      lib.examples.forEach((example) => this.checkDuplicates(project.examples, example));
+      return project;
+    }, project);
+  }
+
+  convert(): MS3.API {
```
Comment (eliasmamo) [inline src/ms3/ms3-to-oas/merge-library-to-ms3.ts]: merge()

File: src/ms3/ms3-to-oas/merge-library-to-ms3.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import * as MS3 from './../ms3-v1-api-interface';
+import { cloneDeep, forEach, map } from 'lodash';
+import { Library } from './../ms3-v1-library-interface';
+import { Trait } from './../ms3-v1-api-interface';
+
+class MergeLibraryToMs3 {
+  libraries: Library[];
```
Comment (eliasmamo) [inline src/ms3/ms3-to-oas/merge-library-to-ms3.ts]: make it private

**Pull request merged**